src/component/feature/strategy/flexible-rollout-strategy.jsx

```diff
@@ -1,28 +1,37 @@
 import React, { Component } from 'react';
 import { Textfield } from 'react-mdl';
+import PropTypes from 'prop-types';
 import strategyInputProps from './strategy-input-props';
 import Select from '../../common/select';
 
 import StrategyInputPercentage from './input-percentage';
 
-const stickinessOptions = [
+const builtInStickinessOptions = [
     { key: 'default', label: 'default' },
     { key: 'userId', label: 'userId' },
     { key: 'sessionId', label: 'sessionId' },
     { key: 'random', label: 'random' },
 ];
 
 export default class FlexibleRolloutStrategy extends Component {
-    static propTypes = strategyInputProps;
+    static propTypes = { ...strategyInputProps, context: PropTypes.array };
 
-    onConfiguUpdate = (field, evt) => {
+    onUpdate = (field, evt) => {
         evt.preventDefault();
         const value = evt.target.value;
         this.props.updateParameter(field, value);
     };
 
+    resolveStickiness = () => {
+        const { context } = this.props;
+        return builtInStickinessOptions.concat(
+            context.filter(c => c.stickiness).map(c => ({ key: c.name, label: c.name }))
```

Comment:
I assume that the c.name is always going to unique here?

Reply:
jup, its a unique constraint in db.

Reply:
...but we should also filter out potential conflicts with values already inside `builtInStickinessOptions`. I will fix